### DISCUSSION TITLE
works flawlessly with Shelly Pro 3EM!

### DISCUSSION THREAD

**Discussion by vormstein** [svenerbe/ecoflow_dynamic_power_adjustment#15] [General]
for the lazy ones: you have to insert the "total active power of your Shelly:

service: pyscript.set_ef_powerstream_custom_load_power
data:
  SerialNumber: XXXXXX
  TotalPower: "{{ states('sensor.shellypro3em_YYYYYYYY_**total_active_power**') | round }}"
  Automation: true

I put the repetition to every second since the data is gathered locally and it works quite well.

**Comment by bragg2012** [comment]
Nice. How is the 1 second refresh going on the automation?  I've got mine at 10 sec now as it was bouncing around all over the place on 5 or less.  Probably a local issue my side with mqtt queues etc and it not being able to keep up.

**Reply by vormstein** [reply]
Yeah, bouncing suits well.. at least it seems like it. It currently works for me (>3hrs 😅) without any issues but I am a bit concerned regarding stability of ecoflow‘s api… will gove feedback after 24hrs

**Reply by vormstein** [reply]
So, coming back from it, the 1s automation does seem to work.

however, I decided to duplicate it and make it work this precise only during „transition times“, i.e. when our solar (12kwp) does not usually provide all of the energy.

to be precise: I will have one automations with 1s interval between 1h before sunset and 1h after sunrise. The other one will be during the day only updating every 5s.
